src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1729,6 +1729,81 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
                 return;
             }
 
+            var usage = ticket.GetUsage();
+            var identifier = ticket.GetProperty(OpenIdConnectConstants.Extra.ClientId);
+            var audiences = ticket.GetAudiences();
+            if (string.Equals(usage, OpenIdConnectConstants.Usages.RefreshToken, StringComparison.Ordinal)) {
+                // For Refresh tokens
+                if ((clientNotification.IsSkipped &&
+                     ticket.ContainsProperty(OpenIdConnectConstants.Extra.ClientAuthenticated)) ||
+                    (!clientNotification.IsSkipped &&
+                     (!ticket.ContainsProperty(OpenIdConnectConstants.Extra.ClientAuthenticated) ||
```

Comment:
I think I recall seeing something in the OAuth 2 spec that said if a client is confidential that they should always have that authentication checked when making requests. If a client is confidential and is trying to check for a refresh token that was given to an unauthenticated client, that's a different client and the authenticated client shouldn't be able to acquire information about the refresh token. Yes, theoretically they could just retry the request as an unauthenticated, but I think it still goes against the intent of refresh tokens only being known to the clients they're given to. I could also see the opposing argument though that if a client is doing scanning attacks that they would first try it as being unauthenticated, but I think it makes it just slightly more troublesome for a client to find out about tokens that weren't issued to them.